File: reference/vector_samples_19_4_10/SIL/SilAdapterBasicPython/CANoe/ApplicationModels/PanelHelper.py
import vector.canoe
from application_layer import RoomTemperatureControl
from application_layer.ExploratoryTesting import Panel

@vector.canoe.measurement_script
class PanelHelper:

    def start(self):
        RoomTemperatureControl.Sensor1.Temperature.register_on_change_handler(self.set_panel_average_temperature)
        RoomTemperatureControl.Sensor2.Temperature.register_on_change_handler(self.set_panel_average_temperature)
        RoomTemperatureControl.Sensor3.Temperature.register_on_change_handler(self.set_panel_average_temperature)
        self.set_panel_average_temperature()

    def set_panel_average_temperature(self):
        Panel.AverageTemperature = (RoomTemperatureControl.Sensor1.Temperature.copy() \
                                  + RoomTemperatureControl.Sensor2.Temperature.copy() \
                                  + RoomTemperatureControl.Sensor3.Temperature.copy()) / 3

File: reference/vector_samples_19_4_10/SIL/SilAdapterBasicCpp/CANoe/ApplicationModels/PanelHelper.cs
using System;
using Vector.Tools;
using Vector.CANoe.Runtime;
using Vector.CANoe.Sockets;
using Vector.CANoe.Threading;
using Vector.Diagnostics;
using NetworkDB;
using NetworkDB._SystemDataTypes;
using RoomTemperatureControl;
using IServiceProvider = Vector.CANoe.Runtime.IServiceProvider;


public class PanelHelper : MeasurementScript
{

    /// <summary>
    /// Called before measurement start to perform necessary initializations,
    /// e.g. to create objects. During measurement, few additional objects
    /// should be created to prevent garbage collection runs in time-critical
    /// simulations.
    /// </summary>
    public override void Initialize()
    {

    }

    /// <summary>Notification that the measurement starts.</summary>
    public override void Start()
    {
    }

    /// <summary>Notification that the measurement ends.</summary>
    public override void Stop()
    {

    }

    /// <summary>
    /// Cleanup after the measurement. Complement to Initialize. This is not
    /// a "Dispose" method; your object should still be usable afterwards.
    /// </summary>
    public override void Shutdown()
    {

    }

    [OnChange(RoomTemperatureControl.Sensor1.MemberIDs.Temperature)]
    [OnChange(RoomTemperatureControl.Sensor2.MemberIDs.Temperature)]
    [OnChange(RoomTemperatureControl.Sensor3.MemberIDs.Temperature)]
    public void OnTemperatureChange()
    {
        SetPanelAverageTemperature();
    }

    public void SetPanelAverageTemperature()
    {
        ExploratoryTesting.Panel.AverageTemperature.Value = (RoomTemperatureControl.Sensor1.Temperature.Value +
                                                RoomTemperatureControl.Sensor2.Temperature.Value +
                                                RoomTemperatureControl.Sensor3.Temperature.Value) / 3.0;
    }
}